android/src/app/organicmaps/routing/LanesAdapter.java

```diff
@@ -0,0 +1,90 @@
+package app.organicmaps.routing;
+
+import android.content.res.ColorStateList;
+import android.util.TypedValue;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+import app.organicmaps.R;
+import app.organicmaps.widget.ArrowView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LanesAdapter extends RecyclerView.Adapter<LanesAdapter.LanesViewHolder>
+{
+  @NonNull
+  private final List<SingleLaneInfo> mItems = new ArrayList<>();
+
+  @NonNull
+  @Override
+  public LanesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
+  {
+    LayoutInflater inflater = LayoutInflater.from(parent.getContext());
```

Comment:
nit: Marking all non-modified variables as `final` helps reviewers to read the code.